examples/mesh_stereo.py

```diff
@@ -0,0 +1,256 @@
+#!/usr/bin/env python3
+
+import cv2
+import numpy as np
+import depthai as dai
+import argparse
+
+"""
+If one or more of the additional depth modes (lrcheck, extended, subpixel)
+are enabled, then:
+ - depth output is FP16. TODO enable U16.
+ - median filtering is disabled on device. TODO enable.
+ - with subpixel, either depth or disparity has valid data.
+
+Otherwise, depth output is U16 (mm) and median is functional.
+But like on Gen1, either depth or disparity has valid data. TODO enable both.
+"""
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "-res",
+    "--resolution",
+    type=str,
+    default="720",
+    help="Sets the resolution on mono cameras. Options: 800 | 720 | 400",
+)
+parser.add_argument(
+    "-md",
+    "--mesh_dir",
+    type=str,
+    default=None,
+    help="Output directory for mesh files. If not specified mesh files won't be saved",
+)
+parser.add_argument(
+    "-static",
+    "--static_frames",
+    default=False,
+    action="store_true",
+    help="Run stereo on static frames passed from host 'dataset' folder",
+)
+args = parser.parse_args()
+
+resolutionMap = {"800": (1280, 800), "720": (1280, 720), "400": (640, 400)}
+
+if args.resolution not in resolutionMap:
+    print("Unsupported resolution, 720P will be used")
+    args.resolution = "720"
+
+resolution = resolutionMap[args.resolution]
+
+mesh_directory = args.mesh_dir  # Output dir for mesh files
+
+# StereoDepth config options. TODO move to command line options
+source_camera = not args.static_frames
+out_depth = False  # Disparity by default
+out_rectified = True  # Output and display rectified streams
+lrcheck = True  # Better handling for occlusions
+extended = False  # Closer-in minimum depth, disparity range is doubled
+subpixel = True  # Better accuracy for longer distance, fractional disparity 32-levels
+# Options: MEDIAN_OFF, KERNEL_3x3, KERNEL_5x5, KERNEL_7x7
+median = dai.StereoDepthProperties.MedianFilter.KERNEL_7x7
+
+# Sanitize some incompatible options
+if lrcheck or extended or subpixel:
+    median = dai.StereoDepthProperties.MedianFilter.MEDIAN_OFF  # TODO
+
+print("StereoDepth config options:")
+print("    Left-Right check:  ", lrcheck)
+print("    Extended disparity:", extended)
+print("    Subpixel:          ", subpixel)
+print("    Median filtering:  ", median)
+
+
+def create_stereo_pipeline(calibData, mesh_directory, from_camera=True):
+    print("Creating Stereo Depth pipeline: ", end="")
+    if from_camera:
+        print("MONO CAMS -> STEREO -> XLINK OUT")
+    else:
+        print("XLINK IN -> STEREO -> XLINK OUT")
+    pipeline = dai.Pipeline()
+
+    if from_camera:
+        cam_left = pipeline.createMonoCamera()
+        cam_right = pipeline.createMonoCamera()
+    else:
+        cam_left = pipeline.createXLinkIn()
+        cam_right = pipeline.createXLinkIn()
```

Comment:
For reading from the host I don't think you added read from files. SO may be remove this